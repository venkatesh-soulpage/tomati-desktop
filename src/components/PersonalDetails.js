import React from "react";
import Form from "react-bootstrap/Form";
import Button from "react-bootstrap/Button";
import PasswordTextField from "components/PasswordTextField";

function PersonalDetails({ values, handleChange, setValues, handleStep }) {
  return (
    <>
      <Form.Group>
        <Form.Control
          type="text"
          placeholder="FullName"
          value={values.full_name}
          onChange={handleChange("full_name")}
          required
        />
      </Form.Group>
      <Form.Group>
        <Form.Control
          type="text"
          placeholder="Company Name"
          value={values.company_name}
          onChange={handleChange("company_name")}
          required
        />
      </Form.Group>
      <Form.Group>
        <Form.Control
          type="email"
          placeholder="Email Address"
          value={values.email}
          onChange={handleChange("email")}
          required
        />
      </Form.Group>
      <Form.Group>
        <PasswordTextField
          value={values.password}
          onChange={handleChange("password")}
          placeholder="Password"
          required
        />
      </Form.Group>
      <Form.Group>
        <Form.Check
          type="checkbox"
          label="I agree to Tomati Terms and Privacy Policy."
          required
        />
      </Form.Group>
      <Form.Group>
        <Form.Check
          type="checkbox"
          label="I agree to receive Tomati news and updates."
        />
      </Form.Group>

      <Form.Group className="d-flex justify-content-end">
        <Button
          form="personal-form"
          type="submit"
          // onClick={handleStep("step", 2)}
          className="btn btn-primary mt-3"
          style={{ borderRadius: "20px" }}
        >
          Continue
        </Button>
      </Form.Group>
    </>
  );
}

export default PersonalDetails;
