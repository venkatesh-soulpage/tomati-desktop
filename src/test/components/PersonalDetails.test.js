import React from "react";
import { render } from "@testing-library/react";
import PersonalDetails from "components/PersonalDetails";
import { Router } from "react-router-dom";
import history from "utils/history";
import { Provider } from "react-redux";
import configureStore from "store";

describe("PersonalDetails page", () => {
  it("matches snapshot", () => {
    const store = configureStore();
    const { asFragment } = render(
      <Provider store={store}>
        <Router history={history}>
          <PersonalDetails
            handleChange={() => {}}
            handleEmailCheck={() => {}}
            setValues={() => {}}
          />
        </Router>
      </Provider>,
      {}
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
