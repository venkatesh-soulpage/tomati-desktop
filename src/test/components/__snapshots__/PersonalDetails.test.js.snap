// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonalDetails page matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="form-group"
  >
    <input
      class="form-control"
      disabled=""
      placeholder="Email Address"
      required=""
      type="email"
      value=""
    />
  </div>
  <div
    class="form-group"
  >
    <input
      class="form-control"
      placeholder="First Name"
      required=""
      type="text"
      value=""
    />
  </div>
  <div
    class="form-group"
  >
    <input
      class="form-control"
      placeholder="Last Name"
      required=""
      type="text"
      value=""
    />
  </div>
  <div
    class="form-group"
  >
    <input
      class="form-control"
      placeholder="Phone Number"
      required=""
      type="number"
      value=""
    />
  </div>
  <div
    class="form-group"
  >
    <div
      class="rdt"
    >
      <input
        class="form-control"
        placeholder="Date of Birth"
        type="text"
        value=""
      />
      <div
        class="rdtPicker"
      >
        <div
          class="rdtDays"
        >
          <table>
            <thead>
              <tr>
                <th
                  class="rdtPrev"
                >
                  <span>
                    ‹
                  </span>
                </th>
                <th
                  class="rdtSwitch"
                  colspan="5"
                  data-value="2"
                >
                  March 2021
                </th>
                <th
                  class="rdtNext"
                >
                  <span>
                    ›
                  </span>
                </th>
              </tr>
              <tr>
                <th
                  class="dow"
                >
                  Su
                </th>
                <th
                  class="dow"
                >
                  Mo
                </th>
                <th
                  class="dow"
                >
                  Tu
                </th>
                <th
                  class="dow"
                >
                  We
                </th>
                <th
                  class="dow"
                >
                  Th
                </th>
                <th
                  class="dow"
                >
                  Fr
                </th>
                <th
                  class="dow"
                >
                  Sa
                </th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td
                  class="rdtDay rdtOld"
                  data-month="1"
                  data-value="28"
                  data-year="2021"
                >
                  28
                </td>
                <td
                  class="rdtDay"
                  data-month="2"
                  data-value="1"
                  data-year="2021"
                >
                  1
                </td>
                <td
                  class="rdtDay"
                  data-month="2"
                  data-value="2"
                  data-year="2021"
                >
                  2
                </td>
                <td
                  class="rdtDay"
                  data-month="2"
                  data-value="3"
                  data-year="2021"
                >
                  3
                </td>
                <td
                  class="rdtDay"
                  data-month="2"
                  data-value="4"
                  data-year="2021"
                >
                  4
                </td>
                <td
                  class="rdtDay"
                  data-month="2"
                  data-value="5"
                  data-year="2021"
                >
                  5
                </td>
                <td
                  class="rdtDay"
                  data-month="2"
                  data-value="6"
                  data-year="2021"
                >
                  6
                </td>
              </tr>
              <tr>
                <td
                  class="rdtDay"
                  data-month="2"
                  data-value="7"
                  data-year="2021"
                >
                  7
                </td>
                <td
                  class="rdtDay"
                  data-month="2"
                  data-value="8"
                  data-year="2021"
                >
                  8
                </td>
                <td
                  class="rdtDay"
                  data-month="2"
                  data-value="9"
                  data-year="2021"
                >
                  9
                </td>
                <td
                  class="rdtDay"
                  data-month="2"
                  data-value="10"
                  data-year="2021"
                >
                  10
                </td>
                <td
                  class="rdtDay"
                  data-month="2"
                  data-value="11"
                  data-year="2021"
                >
                  11
                </td>
                <td
                  class="rdtDay"
                  data-month="2"
                  data-value="12"
                  data-year="2021"
                >
                  12
                </td>
                <td
                  class="rdtDay"
                  data-month="2"
                  data-value="13"
                  data-year="2021"
                >
                  13
                </td>
              </tr>
              <tr>
                <td
                  class="rdtDay"
                  data-month="2"
                  data-value="14"
                  data-year="2021"
                >
                  14
                </td>
                <td
                  class="rdtDay"
                  data-month="2"
                  data-value="15"
                  data-year="2021"
                >
                  15
                </td>
                <td
                  class="rdtDay"
                  data-month="2"
                  data-value="16"
                  data-year="2021"
                >
                  16
                </td>
                <td
                  class="rdtDay"
                  data-month="2"
                  data-value="17"
                  data-year="2021"
                >
                  17
                </td>
                <td
                  class="rdtDay"
                  data-month="2"
                  data-value="18"
                  data-year="2021"
                >
                  18
                </td>
                <td
                  class="rdtDay rdtToday"
                  data-month="2"
                  data-value="19"
                  data-year="2021"
                >
                  19
                </td>
                <td
                  class="rdtDay"
                  data-month="2"
                  data-value="20"
                  data-year="2021"
                >
                  20
                </td>
              </tr>
              <tr>
                <td
                  class="rdtDay"
                  data-month="2"
                  data-value="21"
                  data-year="2021"
                >
                  21
                </td>
                <td
                  class="rdtDay"
                  data-month="2"
                  data-value="22"
                  data-year="2021"
                >
                  22
                </td>
                <td
                  class="rdtDay"
                  data-month="2"
                  data-value="23"
                  data-year="2021"
                >
                  23
                </td>
                <td
                  class="rdtDay"
                  data-month="2"
                  data-value="24"
                  data-year="2021"
                >
                  24
                </td>
                <td
                  class="rdtDay"
                  data-month="2"
                  data-value="25"
                  data-year="2021"
                >
                  25
                </td>
                <td
                  class="rdtDay"
                  data-month="2"
                  data-value="26"
                  data-year="2021"
                >
                  26
                </td>
                <td
                  class="rdtDay"
                  data-month="2"
                  data-value="27"
                  data-year="2021"
                >
                  27
                </td>
              </tr>
              <tr>
                <td
                  class="rdtDay"
                  data-month="2"
                  data-value="28"
                  data-year="2021"
                >
                  28
                </td>
                <td
                  class="rdtDay"
                  data-month="2"
                  data-value="29"
                  data-year="2021"
                >
                  29
                </td>
                <td
                  class="rdtDay"
                  data-month="2"
                  data-value="30"
                  data-year="2021"
                >
                  30
                </td>
                <td
                  class="rdtDay"
                  data-month="2"
                  data-value="31"
                  data-year="2021"
                >
                  31
                </td>
                <td
                  class="rdtDay rdtNew"
                  data-month="3"
                  data-value="1"
                  data-year="2021"
                >
                  1
                </td>
                <td
                  class="rdtDay rdtNew"
                  data-month="3"
                  data-value="2"
                  data-year="2021"
                >
                  2
                </td>
                <td
                  class="rdtDay rdtNew"
                  data-month="3"
                  data-value="3"
                  data-year="2021"
                >
                  3
                </td>
              </tr>
              <tr>
                <td
                  class="rdtDay rdtNew"
                  data-month="3"
                  data-value="4"
                  data-year="2021"
                >
                  4
                </td>
                <td
                  class="rdtDay rdtNew"
                  data-month="3"
                  data-value="5"
                  data-year="2021"
                >
                  5
                </td>
                <td
                  class="rdtDay rdtNew"
                  data-month="3"
                  data-value="6"
                  data-year="2021"
                >
                  6
                </td>
                <td
                  class="rdtDay rdtNew"
                  data-month="3"
                  data-value="7"
                  data-year="2021"
                >
                  7
                </td>
                <td
                  class="rdtDay rdtNew"
                  data-month="3"
                  data-value="8"
                  data-year="2021"
                >
                  8
                </td>
                <td
                  class="rdtDay rdtNew"
                  data-month="3"
                  data-value="9"
                  data-year="2021"
                >
                  9
                </td>
                <td
                  class="rdtDay rdtNew"
                  data-month="3"
                  data-value="10"
                  data-year="2021"
                >
                  10
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </div>
  <div
    class="form-group"
  />
  <div
    class="form-group"
  >
    <div
      class="input-group"
    >
      <input
        class="form-control border-right-0"
        placeholder="Password"
        required=""
        type="password"
        value=""
      />
      <div
        class="input-group-append"
      >
        <div
          style="display: flex; justify-content: center; align-items: center; width: 60px; border: 1px solid #ced4da; border-top-right-radius: 5px; border-bottom-right-radius: 5px; background-color: transparent; cursor: pointer;"
        >
          <small>
            Show
          </small>
        </div>
      </div>
    </div>
  </div>
  <div
    class="input-group"
  >
    <input
      class="form-control border-right-0"
      placeholder="Confirm Password"
      required=""
      type="password"
      value=""
    />
    <div
      class="input-group-append"
    >
      <div
        style="display: flex; justify-content: center; align-items: center; width: 60px; border: 1px solid #ced4da; border-top-right-radius: 5px; border-bottom-right-radius: 5px; background-color: transparent; cursor: pointer;"
      >
        <small>
          Show
        </small>
      </div>
    </div>
  </div>
  <div
    class="d-flex justify-content-end mt-5 form-group"
  >
    <button
      class="btn btn-danger mt-md-3 mt-5 rounded-pill px-4 btn btn-primary"
      form="register-form"
      style="border-radius: 30px; width: 140px; height: 54px;"
      type="submit"
    >
      Continue
    </button>
  </div>
</DocumentFragment>
`;
